### DISCUSSION TITLE
On SpatieMediaLibraryFileUpload,  `deleteAbandonedFiles` can lead to delete valid files

### DISCUSSION THREAD

**Discussion by underdpt** [filamentphp/filament#6325] [Ideas]
Hi,

Consider this use case: an edit form with a file upload, where the uploads are shown on a relation manager. If I upload one file and save the form, it's shown on the relation manager. When I upload another file, the first one is deleted.

Another use case: files can be attached to the model in various ways (p.e. through an API). When uploading manually another file, previously attached files are deleted.

This is because `deleteAbandonedFiles` is called on save, and it's not possible to disable it. I think the developer should be able to disable this behaviour, and also it should be documented.

**Comment by danharrin** [comment]
If you make the file upload `multiple()`, it's not a problem as multiple files can be attached at once. All it does it keep the database in sync with the input, which is the expected behaviour in 95% of cases.

**Comment by underdpt** [comment]
Thanks for your quick reply  @danharrin ,

I think that deleting previously uploaded files should at least be noticed on the documentation. Would you consider a PR to opt-out this behaviour?

Regards
David

**Reply by danharrin** [reply]
Its not possible to opt out. Its media library behaviour to delete files. The model_id column in the media table is not nullable, media has to be associated with a model. I don't think theres much we can change on Filaments side. With our native file upload, files dont get deleted.

**Reply by underdpt** [reply]
Hello,

I think Media Library deletes files when the parent record is deleted. I'm not talking here about deleting the parent record or previous media records attached to the parent, I'm talking about a record that has some media and you use Filament to add more media to it. I don't want to dessociate uploaded files from his parent record or from the media record, I want to maintain previously uploaded files linked to the parent record when uploading new files.

I managed to do it by overloading `\Filament\Forms\Components\SpatieMediaLibraryFileUpload` and disabling `deleteAbandonedFiles()` function, so this is a Filament behaviour. Works for me but I still think this should be documented and it would be nice to allow the developer to disable this functionality.

**Reply by OwKe** [reply]
You could also filter the existing media collection, so they are not included inside `deleteAbandonedFiles()`

```
->filterMediaUsing(function (Collection $media): Collection {
    return $media->where('collection_name', '!=', '...');
})
```
